src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepository.java

```diff
@@ -355,14 +367,22 @@ public void postFinishedComment(
 
     comment = comment.concat(additionalComment);
 
-    this.client.postPullRequestComment(pullRequestId, comment);
+    try {
+      this.client.postPullRequestComment(pullRequestId, comment);
+    } catch (StashApiException e) {
+      logger.info(
+          format(
+              "%s: cannot post Build Finished comment for pull request %s: %s",
+              job.getDisplayName(), pullRequestId, e));
+    }
   }
 
-  public boolean mergePullRequest(String pullRequestId, String version) {
+  public boolean mergePullRequest(String pullRequestId, String version) throws StashApiException {
     return this.client.mergePullRequest(pullRequestId, version);
   }
 
-  private boolean isPullRequestMergeable(StashPullRequestResponseValue pullRequest) {
+  private boolean isPullRequestMergeable(StashPullRequestResponseValue pullRequest)
+      throws StashApiException {
```

Comment:
do we need to add this to the signature?

Reply:
Yes, at least for now. The caller is already logging if `false` is returned. It's better to log from one place.

Eventually we should provide better logging about the reason why a particular PR is not being built, but it's outside the scope of this PR. I'm thinking we could add a page showing the polling status with all PRs seen and the reason why the plugin decided to build or not to build each of them e.g.

```
Last poll at 2019-06-30 9:00
8 pull requests in OPEN state
PR-120 build scheduled
PR-121 already built
PR-122 build queued
PR-123 build running
PR-124 "NO TEST" seen
PR-125 cannot be merged
PR-126 is in conflicted state
PR-127 error checking merge and conflict status
```